source/modulo_components/modulo_components/component.py

```diff
@@ -23,18 +23,21 @@ def __init__(self, node_name: str, *args, **kwargs):
         self.__started = False
         self.__execute_thread = None
         self.add_predicate("is_finished", False)
+        self.add_predicate("in_error_state", False)
```

Comment:
Is `in_error_state` the right name or should it be `on_error`?

Reply:
I was considering this too, but I would prefer to keep it distinct from the on_error transition event that lifecycle components have. If the component stops stepping then it counts as being in an error state.